orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -214,10 +213,13 @@ public static Schema convert(TypeDescription orcSchema) {
         "Error in ORC file, children fields and names do not match.");
 
     List<Types.NestedField> icebergFields = Lists.newArrayListWithExpectedSize(children.size());
-    AtomicInteger lastColumnId = new AtomicInteger(getMaxIcebergId(orcSchema));
-    for (int i = 0; i < children.size(); i++) {
-      icebergFields.add(convertOrcToIceberg(children.get(i), childrenNames.get(i),
-          lastColumnId::incrementAndGet));
+    OrcToIcebergVisitor schemaConverter = new OrcToIcebergVisitor(icebergToOrcMapping("root", orcSchema));
+    for (TypeDescription child : orcSchema.getChildren()) {
+      OrcToIcebergVisitor.visit(child, schemaConverter).ifPresent(icebergFields::add);
```

Comment:
Instead of looping over children here, why not just call the visitor on the root `orcSchema`? Then you could check whether the record was present. If not, throw the exception.

Reply:
The root struct of ORC does not have an Iceberg ID, hence this looping. I've moved this to a `visitSchema` method in the visitor that reuses the code of `visitRecord`.